packages/babel-parser/src/parser/statement.js

```diff
@@ -1784,19 +1765,9 @@ export default class StatementParser extends ExpressionParser {
 
   maybeParseExportDeclaration(node: N.Node): boolean {
     if (this.shouldParseExportDeclaration()) {
-      if (this.isContextual("async")) {
```

Comment:
These checks are redundant because `shouldParseExportDeclaration` will check `isAsyncFunction`
https://github.com/babel/babel/blob/a827ca41f3c07a1ffddeaa00998c772a61a60b68/packages/babel-parser/src/parser/statement.js#L1918-L1938
When `this.isContextual("async")` is true, `isAsyncFunction()` must be true and `async` must be followed by a `function` token, therefore `this.unexpected(next, tt._function);` is never reached.